src/App.tsx

```diff
@@ -1,15 +1,27 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import Title from './components/Title';
 import AddItemForm from './components/AddItemForm';
 import TaskDashboard from './components/TaskDashboard';
+import { Item } from './interfaces/Item';
 
 function App() {
+  const [list, setList] = useState<Item[]>([]);
+
+  const handleAddTask = (taskName: string) => {
+    const newList = [...list];
+    newList.push({
+      id: list.length + 1,
+      name: taskName,
+    });
+    setList(newList);
```

Comment:
aqui voce poderia simplificar , ao inves de criar uma nova list e setar ela aqui , pode-se fazer criando um novo item(que sera adicionado) e o adicionando a lista

```suggestion
    setList([...list,newItem]);
```